axelor-account/src/main/java/com/axelor/apps/account/service/payment/paymentsession/PaymentSessionServiceImpl.java

```diff
@@ -266,16 +266,9 @@ public void retrieveEligibleTerms(PaymentSession paymentSession) throws AxelorEx
             .bind(
                 "pfpValidateStatusPartiallyValidated",
                 InvoiceTermRepository.PFP_STATUS_PARTIALLY_VALIDATED)
-            .fetch();
-
-    eligibleInvoiceTermList = this.filterNotAwaitingPayment(eligibleInvoiceTermList);
-    eligibleInvoiceTermList = this.filterBlocking(eligibleInvoiceTermList, paymentSession);
-    eligibleInvoiceTermList.forEach(
-        invoiceTerm -> {
-          fillEligibleTerm(paymentSession, invoiceTerm);
-          invoiceTermRepository.save(invoiceTerm);
-        });
+            .order("id");
 
+    this.filterInvoiceTerms(eligibleInvoiceTermQuery, paymentSession);
     computeTotalPaymentSession(paymentSession);
```

Comment:
You need to find the paymentSession again before computeTotalPaymentSession, otherwise we get an error sometimes

Reply:
Resolved